pdns/dnsname.hh

```diff
@@ -286,6 +286,42 @@ struct SuffixMatchTree
     }
   }
 
+  void remove(const DNSName &name) const
+  {
+    remove(name.getRawLabels());
+  }
+
+  /* Removes the node at `labels`, also make sure that no empty
+   * children will be left behind in memory
+   */
+  void remove(std::vector<std::string> labels) const
+  {
+    SuffixMatchTree smt(*labels.rbegin());
+    auto child = children.find(smt);
+    if (child == children.end()) {
+      // No subnode found, we're done
+      return;
+    }
+
+    // We have found a child
+    labels.pop_back();
+    if (labels.empty()) {
+      // The child is no longer an endnode
+      child->endNode = false;
+      // TODO clear d_value for this node as d_value = T() would break for types
```

Comment:
I'm not sure that's up to this class to handle that. If we really need to, perhaps we could do something with an optional `boost::optional<T> resetValue=boost::none` parameter and set `d_value = resetValue` when clearing, but that seems brittle.

Reply:
:+1:, I'll remove the comment and keep it like this.